includes/class-wc-gateway-stripe.php

```diff
@@ -578,7 +578,18 @@ public function process_payment( $order_id, $retry = true, $force_save_source =
 				return $this->pre_orders->process_pre_order( $order_id );
 			}
 
-			$prepared_source = $this->prepare_source( get_current_user_id(), $force_save_source );
+			// Check whether there is an existing intent.
+			$intent = $this->get_intent_from_order( $order );
+			if ( isset( $intent->object ) && 'setup_intent' === $intent->object ) {
+				$intent = false; // This function can only deal with *payment* intents
+			}
+
+			$stripe_customer_id = null;
+			if ( $intent ) {
```

Comment:
We should check that `$intent->customer` is set. I know technically it should be set at this point, but we could be defensive just in case.

Reply:
Fixed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1075/commits/5c48e76acf1bfc889bbd4506509dd31b929b0a0b